From c85cc10ac1f2f0f7735c4b053fdb1d0ffb4e36e7 Mon Sep 17 00:00:00 2001
From: Vaibhav Hiremath <hvaibhav@ti.com>
Date: Fri, 18 Nov 2011 19:39:46 +0530
Subject: [PATCH 0399/1092] arm:omap:am33xx: add control_status register addr
 & bit offsets

This information is required for clock tree data,
where we need to read the control_status register
to detect the input Osc freq.
Here control_status register is used for .clksel_reg.

Signed-off-by: Vaibhav Hiremath <hvaibhav@ti.com>
---
 arch/arm/mach-omap2/control.h | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/arch/arm/mach-omap2/control.h b/arch/arm/mach-omap2/control.h
index 0ba68d3..0427cb8 100644
--- a/arch/arm/mach-omap2/control.h
+++ b/arch/arm/mach-omap2/control.h
@@ -338,6 +338,21 @@
 #define AM35XX_HECC_SW_RST		BIT(3)
 #define AM35XX_VPFE_PCLK_SW_RST		BIT(4)
 
+/* AM33XX CONTROL_STATUS bits */
+#define AM33XX_SYSBOOT0			(0xff << 0)
+#define AM33XX_DEVTYPE			(1 << 8)
+#define AM33XX_GPMC_CS0_BW		(1 << 16)
+#define AM33XX_GPMC_CS0_WAITEN		(1 << 17)
+#define AM33XX_GPMC_CS0_ADMUX		(0x3 << 18)
+#define AM33XX_SYSBOOT1			(0x3 << 22)
+
+/*
+ * CONTROL AM33XX STATUS register to identify boot-time configurations
+ */
+#define AM33XX_CONTROL_STATUS_OFF	0x040
+#define AM33XX_CONTROL_STATUS		AM33XX_L4_WK_IO_ADDRESS(AM33XX_CTRL_BASE + \
+						AM33XX_CONTROL_STATUS_OFF)
+
 /*
  * CONTROL OMAP STATUS register to identify OMAP3 features
  */
-- 
1.7.11.2

